### PR TITLE
Guess placed block

### PR DESCRIPTION
**Pull request by nsg**
I and @petterarvidsson talked earlier this week about the possibility to remove the ghost block feature in favour for "block guessing", like most games do. This PR places the selected block (from the belt) at the selected coordinates. The server will still send the block updates and correct possible errors.

The positive:
- This greatly improves the feel of the game.
- Block placements are instant.
- I removed a lot of code

The negative:
- We will lose the ability for plugins to change the behaviour of the secondary mouse button (it will always place a block).
- We will limit the flexibility of the game

Over all, I thing this is a good thing.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -1984,6 +1841,56 @@ void parse_blocks(int p, int q, int k, char* blocks, int size, State *s) {
...
+                    }
+                }
+            }
+        }
+    }
+}
+
+void place_block_global_cords(int x, int y, int z, int w) {
+    place_block(chunked_int(x), chunked_int(z), x, y, z, w);
+}
```
Comment (nsg) [inline src/main.c]: Yes, I know that this function is a little weird. `block_place(...)` is just a copy-past of the inline code from `parse_buffer(...)` and I preferred to keep the changes at a minimum for now. I the future when the B protocol is removed we can re-factor and merge these functions.

**Comment by nsg** [comment]
@petterarvidsson what do you think?

**Comment by petterarvidsson** [comment]
@nsg :+1: This really eliminates any lag! Cool!

**Comment by Henningstone** [comment]
Indeed this is exactly what I thought about with "prediction", I'm happy about it!

**Pull request merged**